File: code-coaches/medium/hex-color-code-generator/hex_color_code_generator.py
def rgb_to_hex(r, g, b):
    # convert each RGB value to a two-digit hex string
    r_hex = format(r, "02x")
    g_hex = format(g, "02x")
    b_hex = format(b, "02x")

    # concatenate the hex strings with a "#" in front
    hex_code = "#" + r_hex + g_hex + b_hex
    return hex_code


def main():
    r_rgb = int(input())
    g_rgb = int(input())
    b_rgb = int(input())

    print(rgb_to_hex(r_rgb, g_rgb, b_rgb))


if __name__ == "__main__":
    main()

File: code-coaches/medium/hex-color-code-generator/hex_color_code_generator.rb
def rgb_to_hex(r, g, b)
  # Convert each RGB value to a two-digit hex string
  r_hex = r.to_s(16).rjust(2, '0')
  g_hex = g.to_s(16).rjust(2, '0')
  b_hex = b.to_s(16).rjust(2, '0')

  # Concatenate the hex strings with a "#" in front
  hex_code = '#' + r_hex + g_hex + b_hex
  return hex_code
end

def main
  r_rgb = gets.chomp.to_i
  g_rgb = gets.chomp.to_i
  b_rgb = gets.chomp.to_i

  puts rgb_to_hex(r_rgb, g_rgb, b_rgb)
end

if __FILE__ == $0
  main
end